### ISSUE TITLE
linter: `typescript/consistent-type-imports` requires multiple fixes to be correct

### ISSUE THREAD

**Issue by JohannesIBK** [oxc-project/oxc#14823]
With the following config:
```json
{
  "typescript/consistent-type-imports": [
    "error",
    {
      "fixStyle": "inline-type-imports",
      "prefer": "type-imports"
    }
  ]
}
```

The following code requires two fixes to be correct. 
```tsx
import { ReactNode } from "react";

export function Component(): ReactNode {
    return <div />
}
```

1. First fix: 
```tsx
import { type ReactNode } from "react";
```
2. Second Fix
```tsx
import type { ReactNode } from "react";
```

If all imports are type imports, I expect the linter to fix it in one step instead of requiring two.

**Comment by camc314** [comment]
When i use this, I don't get the second fix? Do you have a different rule prefering `import type` over `import { type` enabled?

https://github.com/camc314/oxlint-repros/commit/536d1b684ef775e312a1c539c66e13343b807210
```
$ cat index.tsx
import { ReactNode } from 'react';

export function Component(): ReactNode {
    return <div />;
}
```
```
$ cat .oxlintrc.json 
{
    "categories": { "correctness": "off" },
    "rules": {
        "typescript/consistent-type-imports": [
            "error",
            {
                "fixStyle": "inline-type-imports",
                "prefer": "type-imports"
            }
        ]
    }
}
```
```
$ pnpx oxlint --fix
Found 0 warnings and 0 errors.
Finished in 9ms on 1 file using 12 threads.
```
```
$ cat index.tsx
import { type ReactNode } from 'react';

export function Component(): ReactNode {
    return <div />;
}
```

**Comment by JohannesIBK** [comment]
Ok, I thought I had everything else commented out, but apparently not. Yes, you are right, sorry, I had `typescript/no-import-type-side-effects` enabled as well. 

Now this is probably more a feature request than a bug, but would this still be possible?

**Comment by camc314** [comment]
Thanks for confirming. I don't think this is a bug in `typescript/consistent-type-imports`.

With:

```json
{
  "fixStyle": "inline-type-imports",
  "prefer": "type-imports"
}
```

the expected fix from `consistent-type-imports` is:

```ts
import { type ReactNode } from "react";
```

The follow-up rewrite to:

```ts
import type { ReactNode } from "react";
```

comes from `typescript/no-import-type-side-effects`.

If you always prefer top-level `import type` for type-only imports, use `consistent-type-imports` with `"fixStyle": "separate-type-imports"` instead, or omit `fixStyle` since that is the default.

If you want to keep `"inline-type-imports"` generally, while also using `typescript/no-import-type-side-effects` to collapse all-type inline imports, then this needs multiple fix passes. That broader behavior is tracked in #16118.


Thanks!
